OSSM-3251: Add cluster-wide deployment to Service Mesh docs (#58539)
openshift/openshift-docs: modules/ossm-deploy-cluster-wide-control-plane-console.adoc

// Module included in the following assemblies:
//
// * service_mesh/v2x/ossm-create-smcp.adoc

:_content-type: PROCEDURE
[id="ossm-deploy-cluster-wide-control-plane-console_{context}"]
= Configuring the control plane for cluster-wide deployment with the web console

> You can configure the `ServiceMeshControlPlane` resource for cluster-wide deployment using the {product-title} web console.  In this example, `istio-system` is the name of the {SMProductShortName} control plane project.

Reviewer: ```suggestion
You can configure the `ServiceMeshControlPlane` resource for cluster-wide deployment using the {product-title} web console. In this example, `istio-system` is the name of the {SMProductShortName} control plane project.
```

Author: Done

.Prerequisites

> * The {SMProductName} Operator must be installed.

Reviewer: ```suggestion
* The {SMProductName} Operator is installed.
```

Author: Done

> * An account with the `cluster-admin` role.

Reviewer: ```suggestion
* You are logged in using an account with the `cluster-admin` role.
```

Author: Done

.Procedure

> . Log in to the {product-title} web console as a user with the `cluster-admin` role. If you use {product-dedicated}, you must have an account with the `dedicated-admin` role.

Reviewer: ```suggestion
. Log in to the {product-title} web console as a user with the `cluster-admin` role. If you use {product-dedicated}, you must have an account with the `dedicated-admin` role.
```
You already said this in the prereqs. Maybe add the second sentence to the prereqs?

Author: Moved the Openshift dedicated part up to prereqs. Done.

. Create a project named `istio-system`.
+
.. Navigate to *Home* -> *Projects*.
+
.. Click *Create Project*.
+

> .. In the *Name* field, enter `istio-system`. The `ServiceMeshControlPlane` resource must be installed in a project that is separate from your microservices and Operators.

Reviewer 2: Maybe we can say here:
In the *Name* field, enter `istio-system`. *Important*: the `ServiceMeshControlPlane` resource must be installed in a project that is separate from your microservices and Operators.
This is only a suggestion, I think will be more clear for the customers

Author: We discussed. NA due to tooling.

+
These steps use `istio-system` as an example, but you can deploy your {SMProductShortName} control plane in any project as long as it is separate from the project that contains your services.
+
.. Click *Create*.

. Navigate to *Operators* -> *Installed Operators*.

. Click the {SMProductName} Operator, then click *Istio Service Mesh Control Plane*.

. On the *Istio Service Mesh Control Plane* tab, click *Create ServiceMeshControlPlane*.

> . Click *YAML file*.

Reviewer 3: ```suggestion
. Click *YAML view*.
```

Author: Done

> . Create a `ServiceMeshControlPlane` file named `istio-installation.yaml` using the following example. The version of the {SMProductShortName} control plane determines the features available regardless of the version of the Operator.

Reviewer 3: No need to create a file when creating the SMCP via the web console. When you click "YAML view", you are presented with an example YAML that you must then modify/replace.

Author: @luksa I deleted this step and tweaked the one before and after it.

+
.Example version {MaistraVersion} istio-installation.yaml

> [source,yaml]

Reviewer: ```suggestion
+
[source,yaml]
```
- Not sure if this `+`is strictly necessary, but I like to add in the space.

Author: Done

----
apiVersion: maistra.io/v2
kind: ServiceMeshControlPlane
metadata:
  name: basic
  namespace: istio-system
spec:

>   version: v{MaistraVersion}

Reviewer 2: v{MaistraVersion} it's not being replace in doc preview

Author: Looks like it did this time. It now shows 2.4

>    mode: ClusterWide

Author: Done

----

> . Click *Create*. The Operator creates pods, services, and {SMProductShortName} control plane components based on your configuration parameters.

Reviewer 2: We should add here that the operator also creates an SMMR by default if this does not exist with the default configuration 
```
spec:
  memberSelectors:
  - matchLabels:
      istio-injection: enabled
```
Because if the customer sets that label by default in their namespaces will be automatically added to the SMMR file

Author: Done

. To verify the control plane installed correctly, click the *Istio Service Mesh Control Plane* tab.
+

> .. Click the name of the new control plane.

Reviewer 3: ```suggestion
.. Click the name of the new ServiceMeshControlPlane object.
```

Author: Marko and I chatted and decided to use object

> . To verify the control plane installed correctly, click the *Istio Service Mesh Control Plane* tab.

Author: Done

+

> .. Click the *Resources* tab to see the {SMProductName} control plane resources the Operator created and configured.

Reviewer: ```suggestion
.. Click the *Resources* tab to see the {SMProductName} control plane resources that the Operator created and configured.
```

Author: Done
